src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joinSeparator = String(separator);
+    let resultingString = '';
+
+    this.forEach((item, index) => {
```

Comment:
Using Array methods is prohibited in this task. Use loops, index access, and array length if needed.